src/core/utils/components/VirtualScroll.tsx

```diff
@@ -0,0 +1,249 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+const getScrollableParent = (element: HTMLElement | null): HTMLElement => {
+  if (!element || element === document.body) {
+    return document.body;
+  }
+
+  return isElementScrollable(element)
+    ? element
+    : getScrollableParent(element.parentElement);
+};
+
+const isElementScrollable = (element: HTMLElement) => {
+  return /(auto|scroll|overlay)/.test(
+    getElementStyle(element, 'overflow') +
+      getElementStyle(element, 'overflow-y'),
+  );
+};
+
+const getElementStyle = (element: HTMLElement, prop: string) => {
+  return getComputedStyle(element, null).getPropertyValue(prop);
+};
+
+const getElementHeight = (element: HTMLElement | undefined) => {
+  return !!element ? element.getBoundingClientRect().height : 0;
+};
+
+const getCountOfNodesInHeight = (
+  childHeight: number,
+  totalHeight: number,
+  startIndex = 0,
+) => {
+  let i = startIndex;
+  let sum = 0;
+  while (sum < totalHeight) {
+    sum += childHeight;
+    ++i;
+  }
+  i = sum > totalHeight ? i - 1 : i;
+  return i - startIndex;
+};
+
```

Comment:
When `childHeight` is `0` (e.g. empty div), this will produce an infinite loop and hang the browser tab.

This whole function reduces down to `Math.floor(totalHeight / childHeight)`, I think.

Reply:
Ok, so added guard for 0 value and simplified code